libs/server/Resp/RespServerSession.cs

```diff
@@ -175,6 +186,8 @@ internal sealed unsafe partial class RespServerSession : ServerSessionBase
             clusterSession = storeWrapper.clusterProvider?.CreateClusterSession(txnManager, this._authenticator, this._user, sessionMetrics, basicGarnetApi, networkSender, logger);
             clusterSession?.SetUser(this._user);
 
+            this.runningWithAOFWaitForCommitMode = storeWrapper.appendOnlyFile != null && storeWrapper.serverOptions.WaitForCommit;
```

Comment:
no need to introduce another bool variable for this and take up space in the instance. just directly refer to `storeWrapper.appendOnlyFile != null && storeWrapper.serverOptions.WaitForCommit`